govcd/vapp_network_test.go

```diff
@@ -0,0 +1,250 @@
+		}
+	}
+	check.Assert(networkFound, Not(Equals), types.VAppNetworkConfiguration{})
+
+	uuid, err := GetUuidFromHref(networkFound.Link.HREF, false)
+	check.Assert(err, IsNil)
+
+	vappNetwork, err := vapp.GetVappNetworkById(uuid, true)
+	check.Assert(err, IsNil)
+	check.Assert(vappNetwork, NotNil)
+
+	vappNetwork, err = vapp.GetVappNetworkByName(networkName, false)
+	check.Assert(err, IsNil)
+	check.Assert(vappNetwork, NotNil)
+
+	//cleanup
+	task, err := vapp.Delete()
+	check.Assert(err, IsNil)
+	err = task.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+	check.Assert(task.Task.Status, Equals, "success")
+}
+
+func (vcd *TestVCD) Test_UpdateNetworkNatRules(check *C) {
+	vapp, networkName, vappNetworkConfig, err := vcd.prepareVappWithVappNetwork(check, "Test_UpdateNetworkNatRules")
+	check.Assert(err, IsNil)
+
+	networkFound := types.VAppNetworkConfiguration{}
+	for _, networkConfig := range vappNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == networkName {
+			networkFound = networkConfig
+		}
+	}
+	check.Assert(networkFound, Not(Equals), types.VAppNetworkConfiguration{})
+
+	desiredNetConfig := types.NetworkConnectionSection{}
+	desiredNetConfig.PrimaryNetworkConnectionIndex = 0
+	desiredNetConfig.NetworkConnection = append(desiredNetConfig.NetworkConnection,
+		&types.NetworkConnection{
+			IsConnected:             true,
+			IPAddressAllocationMode: types.IPAllocationModePool,
+			Network:                 "Test_UpdateNetworkNatRules_network",
+			NetworkConnectionIndex:  0,
+		})
+
+	// Get org and vdc
+	org, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	vdc, err := org.GetVDCByName(vcd.config.VCD.Vdc, false)
+	check.Assert(err, IsNil)
+	check.Assert(vdc, NotNil)
+
+	// Find catalog and catalog item
+	catalog, err := org.GetCatalogByName(vcd.config.VCD.Catalog.Name, false)
+	check.Assert(err, IsNil)
+	check.Assert(catalog, NotNil)
+	catalogItem, err := catalog.GetCatalogItemByName(vcd.config.VCD.Catalog.CatalogItem, false)
+	check.Assert(err, IsNil)
+	vappTemplate, err := catalogItem.GetVAppTemplate()
+	check.Assert(err, IsNil)
+
+	vm, err := spawnVM("FirstNode", *vdc, *vapp, desiredNetConfig, vappTemplate, check, false, false)
+	check.Assert(err, IsNil)
+
+	vm2, err := spawnVM("SecondNode", *vdc, *vapp, desiredNetConfig, vappTemplate, check, false, false)
+	check.Assert(err, IsNil)
+
+	uuid, err := GetUuidFromHref(networkFound.Link.HREF, false)
+	check.Assert(err, IsNil)
+
+	result, err := vapp.UpdateNetworkNatRules(uuid, []*types.NatRule{&types.NatRule{VMRule: &types.NatVMRule{
+		ExternalPort: -1, InternalPort: 22, VMNicID: 0,
+		VAppScopedVMID: vm.VM.VAppScopedLocalID, Protocol: "TCP"}},
+		&types.NatRule{VMRule: &types.NatVMRule{
+			ExternalPort: 80, InternalPort: 22, VMNicID: 0,
+			VAppScopedVMID: vm2.VM.VAppScopedLocalID, Protocol: "UDP"}}},
+		"portForwarding", "allowTraffic")
+	check.Assert(err, IsNil)
+	check.Assert(result, NotNil)
+
+	// verify
```

Comment:
We should check that `result.Configuration.Features.NatService.NatRule` has at least 2 elements